airbyte-config/models/src/main/java/io/airbyte/config/helpers/CloudLogs.java

```diff
@@ -51,8 +51,35 @@
   List<String> tailCloudLog(Configs configs, String logPath, int numLines) throws IOException;
 
   /**
-   * @return true if configuration is not set
+   * @return true if no cloud logging configuration is set;
    */
-  boolean hasEmptyConfigs(Configs configs);
+  static boolean hasEmptyConfigs(Configs configs) {
+    return !hasAwsCredentials(configs) && !hasGcpCredentials(configs);
+  }
+
+  static CloudLogs createCloudLogClient(Configs configs) {
+    // check if the configs exists, and pick a client.
+    if (hasAwsCredentials(configs)) {
+      return new S3Logs();
+    }
+
+    if (hasGcpCredentials(configs)) {
+      return new GcsLogs();
+    }
+
+    throw new RuntimeException("Error no cloud credentials configured..");
+  }
+
+  private static boolean hasAwsCredentials(Configs configs) {
```

Comment:
This is testing for more than just creds. it is also testing for bucket configuration

Reply:
changed name to reflect testing for general configuration